### ISSUE TITLE
run queries in parallel

### ISSUE THREAD

**Issue by jthandy** [analyst-collective/dbt#62]
`dbt run` should issues multiple commands to the database in parallel. it should use the dependency graph to determine what tasks don't depend on each other and run them as soon as possible (with some configurable max # of concurrent tasks).

this has the potential to ​*significantly*​ speed up `dbt run` execution time, which is really important for usability.

**Comment by drewbanin** [comment]
We could process each subgraph concurrently. Simple, but pretty naive... This will still process nodes within each subgraph serially. See [here](https://networkx.github.io/documentation/networkx-1.9.1/reference/generated/networkx.algorithms.components.connected.connected_component_subgraphs.html#networkx.algorithms.components.connected.connected_component_subgraphs)

What we really want is a topological ordering of nodes in which "ties" are clumped together. Every node that is "tied" can be run at the same time. This could probably work well with a thread pool

**Comment by jthandy** [comment]
If what you're saying is:
- now, do naive version
- later, do better version

...then I'm good with that. Just make sure to make another issue for the better version at the completion of this version.
